controller/cache/cluster.go

```diff
@@ -98,6 +101,11 @@ func (c *clusterInfo) ensureSynced() error {
 		return nil
 	}
 	c.log.Info("Start syncing cluster")
+	appInstanceLabel, err := c.db.GetAppInstanceLabelKey()
```

Comment:
Yes, I was thinking about doing that. But it meant expanding controller's permissions to `watch` all configmaps/secrets in the namespace, instead of getting a single configmap. But it is probably necessary. Will make the change tonight.